src/engine.cpp

```diff
@@ -16,54 +16,69 @@ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLI
 #include <fstream>
 #include <functional>
 #include <random>
+#include <filesystem>
+
 //Create definitions of the RandomEngine class here
 //Methods:
 RandomEngine::RandomEngine():length((uint8_t)8),symb(false),mod(symb?73:63){
     //Reading 100 bytes of data from urandom file for seed generation.
     //this part is independent of user's input, hence defined in the default contructor.
     std::ifstream file("/dev/urandom", std::ios::in | std::ios::binary);
     if(!file){
-    std::cout << "Error opening file\n";
+    std::cout<<"Error opening file\n";
     }
     file.read(seq, sizeof(char)*100);
     file.close();
     std::seed_seq seed(seq, seq+100);
     gen= new std::mt19937_64(seed);
     dist = new std::uniform_int_distribution<int64_t>(0,mod);
+    if(!gen || !dist){
```

Comment:
this makes sense only in theory. In reality I don't think these two can ever fail to instantiate as of now in our code. Can you reproduce a situation when these two are null?

Reply:
Not unless the user explicitly defines them as null. My closest guess is maybe the user does not include the ```<random>``` header but in that case the program just wouldn't compile.

Reply:
We don't need to assume extreme cases like these. If someone starts deleting crucial code then its not our problem. What I mean is that is there a situation where these fail to instantiate with the code as it and I don't think there is.

Reply:
The most we should do in my opinion is to not define destructors but instead use smart pointers to avoid memory leaks or best not use pointers at all if possible. People generally avoid using `new` and `delete` and defining own destructors as they are the source of many memory related problems.

Reply:
new operator might fail in case of disk error like disk full error.

Reply:
I don't see how.

Reply:
The only way I see that ```new``` operator can fail is when the main memory is almost full(which might occur when running heavy software), and the system is not able to allocate the required space, so it returns NULL. But I guess that this **out of memory error** takes place in extreme conditions. Still, every edge test case should be handled.

Reply:
We don't need to do this for applications.